modello-maven-plugin/src/main/java/org/codehaus/modello/maven/ModelloXsdMojo.java

```diff
@@ -45,6 +45,15 @@
     @Parameter( defaultValue = "${project.build.directory}/generated-site/resources/xsd", required = true )
     private File outputDirectory;
 
+    /**
+     * Boolean flag relaxing XSD generation with regards to mandatory elements.
+     * If set to {@code true} will not require mandatory elements in the XML which can be useful if the XML is post processed (e.g. POM merging with parents).
+     * The default value is true for backwards compatibility reasons, but should be set to false for most cases.
+     * @since 2.1.0
+     */
+    @Parameter( defaultValue = "true")
+    private boolean areMandatoryElementsNotEnforced;
```

Comment:
This does not look very usual to have a boolean property starting with `is` or `are`.
Usually, the getters are written `isXxx` but the field is most often named without the `is`/`are` prefix.
I would propose to invert it and name it `enforceMandatoryElements` and default it to `false`.

Reply:
I can rename but the default should be to enforce mandatory elements. The POM case is really an unusual edge case IMHO

Reply:
I may have missed something, but the current default value is `true` for _not enforcing mandatory elements_.  I'm fine with changing the default value though.

Reply:
The javadoc should be changed accordingly if the default value is to be changed.

Reply:
Sorry, my bad. I will think again about defaults.

Reply:
Fixed as you proposed @gnodet in https://github.com/codehaus-plexus/modello/pull/263/commits/9ba65d849e1afaa7449980c2907900341dde816f.

Reply:
Lets start with the default being no enforcement for backwards-compatibility.